adapter.go

```diff
@@ -148,6 +121,9 @@ func (a *Adapter) srcStreamAt(key string, off int64, n int64) (io.ReadCloser, er
 			if errors.Is(err, syscall.ENOENT) {
 				a.sizeCache.Add(key, int64(-1))
 			}
+			if errors.Is(err, io.EOF) {
```

Comment:
Just a remark: I had removed this part because I thought it made no sense to have an EOF when creating a Reader (in this context).

Reply:
I'm not sure it would be actually used with a real-world adapter, but the test suite does return an EOF for streamAt in some cases, and the check is needed in that case